code/model/modelanim.cpp

```diff
@@ -22,70 +22,67 @@ SCP_vector<triggered_rotation> Triggered_rotations;
+	{AnimationTriggerType::Scripted, "scripted"},
+	{AnimationTriggerType::TurretFired, "turret-fired"}
 };
 
-int model_anim_match_type(const char* p)
+AnimationTriggerType model_anim_match_type(const char* p)
 {	
-	int i;
-
 	// standard match
-	for (i = 0; i < MAX_TRIGGER_ANIMATION_TYPES; i++) {
-		if ( !strnicmp(p, Animation_type_names[i], strlen(Animation_type_names[i])) )
-			return i;
+	for (const auto& entry: Animation_type_names) {
+		if (!strnicmp(p, entry.second.c_str(), entry.second.length()))
+			return entry.first;
 	}
 
 	// Goober5000 - misspelling
 	if ( !strnicmp(p, "inital", 6) || !strnicmp(p, "\"inital\"", 8) ) {
 		Warning(LOCATION, "Spelling error in table file.  Please change \"inital\" to \"initial\".");
-		return TRIGGER_TYPE_INITIAL;
+		return AnimationTriggerType::Initial;
 	}
 
 	// Goober5000 - deprecation
 	if ( !strnicmp(p, "docking", 7) || !strnicmp(p, "\"docking\"", 9) ) {
-		mprintf(("The \"docking\" animation type name is deprecated.  Specify \"%s\" instead.\n", Animation_type_names[TRIGGER_TYPE_DOCKING_STAGE_2]));
-		return TRIGGER_TYPE_DOCKING_STAGE_2;
+		mprintf(("The \"docking\" animation type name is deprecated.  Specify \"docking-stage-2\" instead.\n"));
+		return AnimationTriggerType::Docking_Stage2;
 	} else if ( !strnicmp(p, "primary_bank", 12) || !strnicmp(p, "\"primary_bank\"", 14) ) {
-		mprintf(( "The \"primary_bank\" animation type name is deprecated.  Specify \"%s\" instead.\n", Animation_type_names[TRIGGER_TYPE_PRIMARY_BANK]));
-		return TRIGGER_TYPE_PRIMARY_BANK;
+		mprintf(( "The \"primary_bank\" animation type name is deprecated.  Specify \"primary-bank\" instead.\n"));
+		return AnimationTriggerType::PrimaryBank;
 	} else if ( !strnicmp(p, "secondary_bank", 14) || !strnicmp(p, "\"secondary_bank\"", 16) ) {
-		mprintf(("The \"secondary_bank\" animation type name is deprecated.  Specify \"%s\" instead.\n", Animation_type_names[TRIGGER_TYPE_SECONDARY_BANK]));
-		return TRIGGER_TYPE_SECONDARY_BANK;
+		mprintf(("The \"secondary_bank\" animation type name is deprecated.  Specify \"secondary-bank\" instead.\n"));
+		return AnimationTriggerType::SecondaryBank;
 	} else if ( !strnicmp(p, "door", 4) || !strnicmp(p, "\"door\"", 6) ) {
-		mprintf(("The \"door\" animation type name is deprecated.  Specify \"%s\" instead.\n", Animation_type_names[TRIGGER_TYPE_DOCK_BAY_DOOR]));
-		return TRIGGER_TYPE_DOCK_BAY_DOOR;
+		mprintf(("The \"door\" animation type name is deprecated.  Specify \"fighterbay\" instead.\n"));
+		return AnimationTriggerType::DockBayDoor;
 	} else if ( !strnicmp(p, "turret firing", 13) || !strnicmp(p, "\"turret firing\"", 15) ) {
-		mprintf(("The \"turret firing\" animation type name is deprecated.  Specify \"%s\" instead.\n", Animation_type_names[TRIGGER_TYPE_TURRET_FIRING]));
-		return TRIGGER_TYPE_TURRET_FIRING;
+		mprintf(("The \"turret firing\" animation type name is deprecated.  Specify \"turret-firing\" instead.\n"));
+		return AnimationTriggerType::TurretFiring;
```

Comment:
Why remove the `Animation_type_names` argument?  That removes any chance of misspelling (not that you misspelled anything), and automatically updates the warning if for some reason the name changes in the future.